Remove usage of cider.nrepl.middleware.util.misc/err-info

N.b. I realize this is an issue against unstable release, just a heads-up that this will break for future stable releases without a fix.

`err-info` was removed in this commit: 
https://github.com/clojure-emacs/cider-nrepl/commit/76f672201997ceed8d79a6427bcf137c32739a04

`cider-jack-in` with unstable clj-refactor.el installed will inject the latest snapshot of cider-nrepl, which breaks refactor-nrepl.

## Expected behavior
This manifests in `cider-jack-in` in my case: I expected it to work, and start a REPL.
## Actual behavior
An exception is thrown, with the following stacktrace:
```
error in process sentinel: Could not start nREPL server: Error loading refactor-nrepl.middleware: java.lang.IllegalAccessError: err-info does not exist, compiling:(refactor_nrepl/middleware.clj:1:1)
Exception in thread "main" java.lang.RuntimeException: Unable to resolve var: refactor-nrepl.middleware/wrap-refactor in this context, compiling:(/tmp/form-init3503199879813880830.clj:1:6826)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6875)
	at clojure.lang.Compiler.analyze(Compiler.java:6669)
	at clojure.lang.Compiler.analyze(Compiler.java:6625)
	at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3834)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6870)
	at clojure.lang.Compiler.analyze(Compiler.java:6669)
	at clojure.lang.Compiler.analyze(Compiler.java:6625)
	at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3834)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6870)
	at clojure.lang.Compiler.analyze(Compiler.java:6669)
	at clojure.lang.Compiler.access$300(Compiler.java:38)
	at clojure.lang.Compiler$LetExpr$Parser.parse(Compiler.java:6269)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6868)
	at clojure.lang.Compiler.analyze(Compiler.java:6669)
	at clojure.lang.Compiler.analyze(Compiler.java:6625)
	at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:6001)
	at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5380)
	at clojure.lang.Compiler$FnExpr.parse(Compiler.java:3972)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6866)
	at clojure.lang.Compiler.analyze(Compiler.java:6669)
	at clojure.lang.Compiler.eval(Compiler.java:6924)
	at clojure.lang.Compiler.eval(Compiler.java:6917)
	at clojure.lang.Compiler.load(Compiler.java:7379)
	at clojure.lang.Compiler.loadFile(Compiler.java:7317)
	at clojure.main$load_script.invokeStatic(main.clj:275)
	at clojure.main$init_opt.invokeStatic(main.clj:277)
	at clojure.main$init_opt.invoke(main.clj:277)
	at clojure.main$initialize.invokeStatic(main.clj:308)
	at clojure.main$null_opt.invokeStatic(main.clj:342)
	at clojure.main$null_opt.invoke(main.clj:339)
	at clojure.main$main.invokeStatic(main.clj:421)
	at clojure.main$main.doInvoke(main.clj:384)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.RuntimeException: Unable to resolve var: refactor-nrepl.middleware/wrap-refactor in this context
	at clojure.lang.Util.runtimeException(Util.java:221)
	at clojure.lang.Compiler$TheVarExpr$Parser.parse(Compiler.java:710)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6868)
	... 36 more
Subprocess failed
```
## Steps to reproduce the problem
1 ) lein new foo
2) Open foo/project.clj in Emacs, M-x `cider-jack-in`

## Environment & Version information

### clj-refactor.el and refactor-nrepl version information
clj-refactor 2.4.0-SNAPSHOT (package: 20171117.317), refactor-nrepl 2.4.0-SNAPSHOT

### CIDER version information
CIDER 0.17.0snapshot (package: 20180110.533)

### Leiningen or Boot version
Leiningen 2.8.1 on Java 1.8.0_144 OpenJDK 64-Bit Server VM

### Emacs version
GNU Emacs 25.3.1 (x86_64-pc-linux-gnu, GTK+ Version 3.22.26) of 2017-12-04

### Operating system
Arch Linux

--- a/src/refactor_nrepl/middleware.clj
+++ b/src/refactor_nrepl/middleware.clj
@@ -1,6 +1,6 @@
 (ns refactor-nrepl.middleware
   (:require [cider.nrepl.middleware.util.cljs :as cljs]
-            [cider.nrepl.middleware.util.misc :refer [err-info]]
+            [clojure.stacktrace :refer [print-cause-trace]]
             [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
             [clojure.tools.nrepl.misc :refer [response-for]]
             [clojure.tools.nrepl.transport :as transport]
@@ -13,6 +13,12 @@
 (defn- require-and-resolve [sym]
   (require (symbol (namespace sym)))
   (resolve sym))
+
+(defn err-info
+  [ex status]
+  {:ex (str (class ex))
+   :err (with-out-str (print-cause-trace ex))
+   :status #{status :done}})
 
 (defmacro ^:private with-errors-being-passed-on [transport msg & body]
   `(try

